libpod/image/image.go

```diff
@@ -131,41 +142,72 @@ func (ir *Runtime) NewFromLocal(name string) (*Image, error) {
 	return &image, nil
 }
 
+// DefaultPullConfig returns a default new image config for runtime.New()
+func (ir *Runtime) DefaultPullConfig(name string) *PullConfig {
+	imgCtx := new(PullConfig)
+	imgCtx.Name = name
+	imgCtx.SignaturePolicyPath = ""
+	imgCtx.Authfile = ""
+	imgCtx.Writer = nil
+	imgCtx.DockerOptions = nil
+	imgCtx.SigningOptions = SigningOptions{}
+	imgCtx.Label = nil
+	return imgCtx
+}
+
 // New creates a new image object where the image could be local
 // or remote
-func (ir *Runtime) New(ctx context.Context, name, signaturePolicyPath, authfile string, writer io.Writer, dockeroptions *DockerRegistryOptions, signingoptions SigningOptions, forcePull bool, label *string) (*Image, error) {
+func (ir *Runtime) New(ctx context.Context, config *PullConfig, forcePull, update bool) (*Image, error) {
 	span, _ := opentracing.StartSpanFromContext(ctx, "newImage")
 	span.SetTag("type", "runtime")
 	defer span.Finish()
 
 	// We don't know if the image is local or not ... check local first
 	newImage := Image{
-		InputName:    name,
+		InputName:    config.Name,
 		Local:        false,
 		imageruntime: ir,
 	}
+
+	if config.SignaturePolicyPath == "" {
+		config.SignaturePolicyPath = ir.SignaturePolicyPath
```

Comment:
This modifies the caller-held value and is a bit risky (considering that at least one caller reuses the config across multiple calls). Maybe this should create a copy of `config` before modifying it.

Or push this logic deeper into the call stack, into `ir.pull*`.

(Or, perhaps, even make `GetSystemContext` a method on `Runtime`, or add yet another wrapper? There are quite a few wrappers/helpers already, and maybe that can be untangled/simplified. But that’s probably for another PR.)